src/Illuminate/Testing/Concerns/CreatesApplication.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Illuminate\Testing\Concerns;
+
+use RuntimeException;
+
+trait CreatesApplication
+{
+    /**
+     * The application resolver callback.
+     *
+     * @var \Closure|null
+     */
+    protected static $applicationResolver;
+
+    /**
+     * Set the application resolver callback.
+     *
+     * @param  \Closure|null  $resolver
+     * @return void
+     */
+    public static function resolveApplicationUsing($resolver)
+    {
+        static::$applicationResolver = $resolver;
+    }
+
+    /**
+     * Creates the application.
+     *
+     * @return \Illuminate\Contracts\Foundation\Application
+     *
+     * @throws \RuntimeException
+     */
+    protected function createApplication()
+    {
+        $applicationResolver = static::$applicationResolver ?: function () {
+            if (trait_exists(\Tests\CreatesApplication::class)) {
+                $applicationCreator = new class
+                {
+                    use \Tests\CreatesApplication;
+                };
+
+                return $applicationCreator->createApplication();
+            } elseif (file_exists(getcwd().'/bootstrap/app.php')) {
```

Comment:
Isn't it better to use `base_path('bootstrap/app.php')` instead of `getcwd()` here?

Reply:
I think the issue here is the base path can change depending on your environment. For instance, if testing with testbench, the base path is `vendor/orchestra/testbench-core/laravel`. I think the desired `app.php` file in this case would be from the project itself.

Perhaps @nunomaduro can comment as this is his code from `ParallelRunner`.